analytics.js

```diff
@@ -894,6 +922,27 @@ class Analytics {
       // convert integrations object to server identified names, kind of hack now!
       transformToServerNames(rudderElement.message.integrations);
 
+      // Removing true values from clientSuppliedIntegrations except All
+      const tempClientSuppliedIntegrations = Object.keys(
+        clientSuppliedIntegrations
+      )
+        .filter((integration) => {
+          if (integration !== "All") {
+            return clientSuppliedIntegrations[integration] !== true;
```

Comment:
This logic will also filter other integrations from the client-supplied integration that are set to true. We only want that for tempIntegrationsData i.e. GA4. Please modify the logic.

Reply:
no, this logic is required because if some integrations are set to true and for that if data is required for that integration inside integrationsObj then SDK global object should take priority. we should override default true value and set the data if required this is already discussed with @saikumarrs . this is an edge case that we are handling here?

you can take reference of test cases here :
https://www.notion.so/rudderstacks/Test-Results-3f723a53a98f4f3183a48bfa086afa65

Reply:
@mihir-4116 Correct, but we should do it only for those integrations that have data defined in the global integrations data object. Otherwise, we'll be messing up with the event filtering logic.

Reply:
updated